Packages.yml page (#8433)
fishtown-analytics/docs.getdbt.com: website/docs/reference/dbt-jinja-functions/packages.yml.md

---
title: "About packages.yml context"
sidebar_label: "packages.yml context"
id: "packages.yml context"
description: "Use these context methods to configure dependencies in the packages.yml file."
---

The following context methods and variables are available when configuring a `packages.yml` file. 

**Available context methods:**
- [env_var](/reference/dbt-jinja-functions/env_var)

>     - You can use `env_var()` in any dbt YAML file that supports Jinja, but `packages.yml` treat environment variables as [secrets](/docs/build/dbt-tips#yaml-tips) (using the `DBT_ENV_SECRET_` prefix).

Reviewer: ```suggestion
    - Use `env_var()` in any dbt YAML file that supports Jinja. Only `packages.yml` and `profiles.yml` support environment variables for [secure values](/docs/build/dbt-tips#yaml-tips) (using the `DBT_ENV_SECRET_` prefix).
```

- [var](/reference/dbt-jinja-functions/var) (Note: only variables defined with `--vars` are available. Refer to [YAML tips](/docs/build/dbt-tips#yaml-tips) for more information)

**Available context variables:**
- [builtins](/reference/dbt-jinja-functions/builtins)
- [dbt_version](/reference/dbt-jinja-functions/dbt_version)
- [target](/reference/dbt-jinja-functions/target)

> 

Reviewer: it might be helpful to have an example for the user so they know how to use a method or variable?

---
title: "About packages.yml context"
sidebar_label: "packages.yml context"
id: "packages.yml context"
description: "Use these context methods to configure dependencies in the packages.yml file."
---

The following context methods and variables are available when configuring a `packages.yml` file. 

**Available context methods:**
- [env_var](/reference/dbt-jinja-functions/env_var)
    - Use `env_var()` in any dbt YAML file that supports Jinja. Only `packages.yml` and `profiles.yml` support environment variables for [secure values](/docs/build/dbt-tips#yaml-tips) (using the `DBT_ENV_SECRET_` prefix).
- [var](/reference/dbt-jinja-functions/var) (Note: only variables defined with `--vars` are available. Refer to [YAML tips](/docs/build/dbt-tips#yaml-tips) for more information)

> **Available context variables:**

Reviewer: great examples @nataliefiann - do you think we should add the env_var one without the macro? it's the simplest one that is quite useful. the example you added in your docs plan i mean

Reviewer: thanks for adding the example @nataliefiann! looks good - i have a sugg on my end in terms of the structure of this. it currently doesn't follow the other docs (avail methods and then show the examples under an 'examples' header). i think currently the way this is formatted and laid out, its a bit overwhelming because we're introducing all at once for the user. i'd recommend something like:}

**Available context methods:**
- [env_var](/reference/dbt-jinja-functions/env_var)
    - Use `env_var()` in any dbt YAML file that supports Jinja. Only `packages.yml` and `profiles.yml` support environment variables for [secure values](/docs/build/dbt-tips#yaml-tips) (using the `DBT_ENV_SECRET_` prefix).
- [var](/reference/dbt-jinja-functions/var) (Note: only variables defined with `--vars` are available. Refer to [YAML tips](/docs/build/dbt-tips#yaml-tips) for more information)

**Available context variables:**
- [builtins](/reference/dbt-jinja-functions/builtins)
- [dbt_version](/reference/dbt-jinja-functions/dbt_version)
- [target](/reference/dbt-jinja-functions/target)

## Example usage
Here are some examples that xyz...

**builtin variables**
To use xyz on using a builtins variable.....

```yaml
packages:
  - package: dbt-labs/dbt_utils
    version: "{% if builtins is defined %}0.14.0{% else %}0.13.1{% endif %}"
```

**env_var**
To use xyz on with env_var...

```yaml
packages:
  - package: dbt-labs/dbt_utils
    version: "{{ env_var('DBT_UTILS_VERSION') }}"
```

**dbt_version**
....add dbt version example that includes packages...

**target**
... add target example that includes packages**

## Related docs

- [Packages](/docs/build/packages)

---
title: "About packages.yml context"
sidebar_label: "packages.yml context"
id: "packages.yml context"
description: "Use these context methods to configure dependencies in the packages.yml file."
---

The following context methods and variables are available when configuring a `packages.yml` file. 

**Available context methods:**
- [env_var](/reference/dbt-jinja-functions/env_var)
    - Use `env_var()` in any dbt YAML file that supports Jinja. Only `packages.yml` and `profiles.yml` support environment variables for [secure values](/docs/build/dbt-tips#yaml-tips) (using the `DBT_ENV_SECRET_` prefix).
```

> "{{ env_var('MY_ENV_VAR') }}"

Reviewer: this example is a little unclear, how does the user know how to use this in packages.yml?

- [var](/reference/dbt-jinja-functions/var) (Note: only variables defined with `--vars` are available. Refer to [YAML tips](/docs/build/dbt-tips#yaml-tips) for more information)

**Available context variables:**
- [builtins](/reference/dbt-jinja-functions/builtins)
- [dbt_version](/reference/dbt-jinja-functions/dbt_version)
- [target](/reference/dbt-jinja-functions/target)

## Example usage

The following examples show how to use the different context methods and variables in your `packages.yml`.

Use `builtins` in your `packages.yml`:

```
packages:
  - package: dbt-labs/dbt_utils
    version: "{% if builtins is defined %}0.14.0{% else %}0.13.1{% endif %}"

```

Use `env_var` in your `packages.yml`:

```
packages:
  - package: dbt-labs/dbt_utils
    version: "{{ env_var('DBT_UTILS_VERSION') }}"
```

Use `dbt_version` in your `packages.yml`:

```
packages:
  - package: dbt-labs/dbt_utils
    version: "{% if dbt_version is defined %}0.14.0{% else %}0.13.1{% endif %}"

```

Use `target` in your `packages.yml`:

```

packages:
  - package: dbt-labs/dbt_utils

>     version: "{% if env_var('DBT_ENV_NAME') == 'prod' %}0.14.0{% else %}0.13.1{% endif %}"

Reviewer: hey @nataliefiann , i just saw this before approving - is this the right code? i don't see the `target` variable mentioned in the code - instead you're using env_var?

>     version: "{% if env_var('DBT_ENV_NAME') == 'prod' %}0.14.0{% else %}0.13.1{% endif %}"

Author: ```suggestion
    version: "{% if target.name == 'prod' %}0.14.0{% else %}0.13.1{% endif %}"
```
